src/components/powerInfoPage/datePicker.tsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import classNames from 'classnames';
+
+interface IProps {
+  changeDate(date: Date): void;
+}
+
+const DatePicker: React.FC<IProps> = ({ changeDate }) => {
+  const onChangeDate = (event: React.ChangeEvent<HTMLInputElement>) => {
+    // to prevent user from choosing invalid date cause a api can't fetch problem
+    const lastDate =
+      new Date(event.target.value).getTime() > new Date().getTime()
+        ? new Date()
+        : new Date(event.target.value);
+    changeDate(lastDate);
+  };
+
+  return (
+    <div className={classNames('powerinfo-datepicker-container')}>
+      <input
+        className={classNames('powerinfo-datepicker-input')}
+        type="date"
+        onChange={(event) => onChangeDate(event)}
```

Comment:
I think it's better to set the default value to show current date when user get into this page without choosing a date.
You can complete this by following these steps:
1. Create a variable to get current time from parent component
```tsx
// powerInfo/datePicker.tsx
interface IProps {
  changeDate(date: Date): void;
  currDate: Date;
}

const DatePicker: React.FC<IProps> = ({ changeDate, currDate }) => {
// ...
```
2. Pass the date in parent
```tsx
// powerInfo/index.tsx
<DatePicker changeDate={setDate} currDate={date} />
```
3. Use `dayjs` to format the variable and set to `value`
```tsx
// powerInfo/datePicker.tsx
import dayjs from 'dayjs';
// ...
<input
    className={classNames('powerinfo-datepicker-input')}
    type="date"
    value={dayjs(currDate).format('YYYY-MM-DD')}  // new
    onChange={(event) => onChangeDate(event)}
/>
```

Reply:
Done. Thanks!